### PR TITLE
fix(transport): Include ServerName in TLS Config for GRPC

### PR DESCRIPTION
**Pull request by andyrzhao**
According to Golang TLS documentation, ServerName is used to verify the hostname on the returned certificate, unless InsecureSkipVerify is specified.

### CODE REVIEW

File: transport/grpc/dial.go [hunk 1]
```diff
@@ -156,8 +157,13 @@ func dial(ctx context.Context, insecure bool, o *internal.DialSettings) (*grpc.C
 			}
 			// TODO(cbro): add support for system parameters (quota project, request reason) via chained interceptor.
 		} else {
+			endpointUrl, err := url.Parse(endpoint)
```
Comment (codyoss) [inline transport/grpc/dial.go:160]: nit: `endpointURL`

Comment (andyrzhao) [reply transport/grpc/dial.go]: done

**Comment by andyrzhao** [comment]
Please hold off on Approving. Verifying change with Shin.

**Review by shinfan — APPROVED** [review]

**Comment by andyrzhao** [comment]
Abandoning. More verification needed.

**Pull request closed without merging**